hyper/http20/connection.py

```diff
@@ -170,8 +170,9 @@ def request(self, method, url, body=None, headers={}):
         """
         stream_id = self.putrequest(method, url)
 
+        default_headers = (':method', ':scheme', ':authority', ':path')
         for name, value in headers.items():
-            self.putheader(name, value, stream_id)
+            self.putheader(name, value, stream_id, replace=name in default_headers)
```

Comment:
I'm worried here about the possibility that people may pass the header as a non-native string on the platform (e.g. `unicode` on Python 2 or `bytes` on Python 3). I think we may need to do some coercion of these headers, maybe using a method like [`to_native_string` from requests](https://github.com/kennethreitz/requests/blob/master/requests/utils.py#L690-L705).